anvil/src/main/kotlin/dev/inkremental/Anvil.kt

```diff
@@ -0,0 +1,326 @@
+                }
+                if (removeChildren) {
+                    viewGroup.removeViews(0, childCount)
+                }
+            }
+        }
+    }
+
+    /**
+     * Returns currently rendered Mount point. Must be called from the
+     * Renderable's view() method, otherwise it returns null
+     * @return current mount point
+     */
+    fun currentMount(): Mount? {
+        return currentMount
+    }
+
+    /**
+     * Returns currently rendered View. It allows to access the real view from
+     * inside the Renderable.
+     * @return currently rendered View
+     */
+    fun <T : View?> currentView(): T? {
+        return if (currentMount == null) {
+            null
+        } else currentMount?.iterator?.currentView() as T?
+    }
+
+    fun render(v: View?) {
+        val m = mounts[v] ?: return
+        render(m)
+    }
+
+    fun render(m: Mount) {
+        if (m.lock) {
+            return
+        }
+        m.lock = true
+        val prev = currentMount
+        currentMount = m
+        m.iterator.start()
+        if (m.renderable != null) {
+            m.renderable.view()
+        }
+        m.iterator.end()
+        currentMount = prev
+        m.lock = false
+    }
+
+    //waiting for Kotlin 1.4
+//    /** Renderable can be mounted and rendered using Anvil library.  */
+//    interface Renderable {
+//        /** This method is a place to define the structure of your layout, its view
+//         * properties and data bindings.  */
+//        fun view()
+//    }
+
+    interface ViewFactory {
+        fun fromClass(c: Context?, v: Class<out View?>): View?
+        fun fromXml(parent: ViewGroup, xmlId: Int): View?
+    }
+
+    internal class DefaultViewFactory : ViewFactory {
+        override fun fromClass(c: Context?, viewClass: Class<out View?>): View? {
+            return try {
+                viewClass.getConstructor(Context::class.java).newInstance(c)
+            } catch (e: InvocationTargetException) {
+                throw RuntimeException(e)
+            } catch (e: IllegalAccessException) {
+                throw RuntimeException(e)
+            } catch (e: NoSuchMethodException) {
+                throw RuntimeException(e)
+            } catch (e: InstantiationException) {
+                throw RuntimeException(e)
+            }
+        }
+
+        override fun fromXml(parent: ViewGroup, xmlId: Int): View? {
+            return LayoutInflater.from(parent.context).inflate(xmlId, parent, false)
+        }
+    }
+
+    interface AttributeSetter<T> {
+        operator fun set(v: View, name: String, value: T?, prevValue: T?): Boolean
+    }
+
+    /** Mount describes a mount point. Mount point is a Renderable function
+     * attached to some ViewGroup. Mount point keeps track of the virtual layout
+     * declared by Renderable  */
+    class Mount(v: View, val renderable: Renderable?) {
+        var lock = false
+        private val rootView: WeakReference<View>
```

Comment:
IDE suggests joining this with assignment from the `init` block below. Let's do this.